tests/arch/arm/arm_thread_swap/src/arm_thread_arch.c

```diff
@@ -334,26 +336,44 @@ static void alt_thread_entry(void)
 	 */
 #if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)
 	__asm__ volatile (
-		"push {r4,r5,r6,r7};\n\t"
-		"mov r4, r8;\n\t"
-		"mov r5, r9;\n\t"
-		"push {r4, r5};\n\t"
-		"mov r4, r10;\n\t"
-		"mov r5, r11;\n\t"
-		"push {r4, r5};\n\t"
-		"push {r0, r1};\n\t"
-		"mov r1, r7;\n\t"
-		"mov r0, %0;\n\t"
-		"ldmia r0!, {r4-r7};\n\t"
-		"ldmia r0!, {r4-r7};\n\t"
-		"mov r8, r4;\n\t"
-		"mov r9, r5;\n\t"
-		"mov r10, r6;\n\t"
-		"mov r11, r7;\n\t"
-		"mov r7, r1;\n\t"
-		"pop {r0, r1};\n\t"
-		:	: "r" (&ztest_thread_callee_saved_regs_container)
-		: "memory"
+		/* Stash r4-r11 in stack, they will be restored much later in
+		 * another inline asm -- that should be reworked since stack
+		 * must be balanced when we leave any inline asm. We could
+		 * use simply an alternative stack for storing them instead
+		 * of the function's stack.
+		 */
+		"push {r4-r7};"
```

Comment:
@microbuilder Sorry, I missed that, but you need to add `\n\t` in all these lines. I tend to not use them, but to be consist with the rest of the code in the test I think it's good to add them likewise here.